read.go

```diff
@@ -126,11 +122,14 @@ func (r *reader) readTOC(toc *indexTOC) error {
 				return err
 			}
 
+			skipSection := len(tags) > 0 && !slices.Contains(tags, tag)
```

Comment:
Instead of introducing the "skip" concept, I could have taken advantage of the fact that the metadata sections are always first in the TOC. However, our index reading code is structured around flexible "section tags", and I got the feeling that section ordering wasn't an invariant we wanted to rely on.